### DISCUSSION TITLE
Safe parse a stringified object

### DISCUSSION THREAD

**Discussion by christoph-fricke** [colinhacks/zod#367] [Q&A]
I recently discovered zod and absolutely love the library. Great work! :+1:

A few days ago I found a perfect use case where I have to safe parse a string to an object and make sure that it conforms with a required schema. Naturally my first thought was to use zod for this.
```typescript
const schema = z.object({
  logoUrl: z.string().url("Logo has to be a valid URL."),
  appbarBg: z.string().refine(isHexColor),
  elementStatus: z.object({
    status100: z.string().refine(isHexColor),
    status200: z.string().refine(isHexColor),
  }),
```

Turns out that calling `schema.safeParse("Valid object as string")` (String is simplified) fails because a string is not an object.
 Calling `schema.safeParse(JSON.parse("Valid object as string"))` would work but destroys the idea of `safeParse` since `JSON.parse` throws an error for invalid JSON strings.

Is there an option in zod to parse a stringified object? I could find something in the docs. I currently use the version `1.11.13`

**Comment by colinhacks** [comment]
This doesn't exist natively in Zod currently. I recommend parsing the JSON externally to Zod (with your own try/catch logic) then pass the properly parsed JSON into your schema. 

The behavior of `safeParse` is somewhat subtle in that it doesn't catch all errors; for instance, if an error is thrown inside a refinement or transform function, it is thrown normally. `safeParse` is just an easier way to handle ZodErrors, which contain a list of all the validation issues that are detected during parsing.
